tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -44,23 +49,29 @@ def __init__(self, test_context):
 
         si_settings.load_context(self.logger, test_context)
 
+        if self.override_default_topic_replication:
```

Comment:
When I was running this test on repeat, I saw the test fail a few times when waiting on consumer. Looking at the consumer log I saw the following errors repeated (on failing test instances):

```
[2022-04-27 06:30:11,486] WARN [Consumer clientId=consumer-test_group-1, groupId=test_group] Connection to node 2147483646 (docker_n_1/172.18.0.5:9092) could not be established. Broker may not be available. (org.apache.kafka.clients.NetworkClient)
[2022-04-27 06:30:11,486] INFO [Consumer clientId=consumer-test_group-1, groupId=test_group] Group coordinator docker_n_1:9092 (id: 2147483646 rack: null) is unavailable or invalid due to cause: null.isDisconnected: true. Rediscovery will be attempted. (org.apache.kafka.clients.consumer.internals.AbstractCoordinator)
[2022-04-27 06:30:11,486] INFO [Consumer clientId=consumer-test_group-1, groupId=test_group] Rebalance failed. (org.apache.kafka.clients.consumer.internals.AbstractCoordinator)
```

It seems on researching that error, that when group coordinator is killed the tests fail because `__consumer_offsets` is not replicated by default and there is no resilience (please correct me if I am mistaken), so I set replication for all topics by default to number of brokers, after which the tests pass correctly even if the coordinator is killed.

I could not find a specific setting to control the replication for the consumer offsets topic.

I have cleaned up the code so that the replication count is set up in the specific test where it is required and not spread across the base and child classes.